pkg/transforms/transformer.go

```diff
@@ -82,6 +83,32 @@ type Deletion struct {
 	UID string `json:"uid,omitempty"`
 }
 
+func buildSelfLink(nodeProps map[string]interface{}, resourceString string) string {
+	namespace := ""
+	if nodeProps["namespace"] != nil && nodeProps["namespace"].(string) != "" {
+		namespace = "/namespaces/" + nodeProps["namespace"].(string)
+	}
+
+	root := "api/v1"
+	groupVersion := ""
+	if nodeProps["apigroup"] != nil &&
+		nodeProps["apigroup"].(string) != "" &&
+		nodeProps["apiversion"] != nil &&
+		nodeProps["apiversion"].(string) != "" {
+		root = "apis"
+		groupVersion = "/" + nodeProps["apigroup"].(string) + "/" + nodeProps["apiversion"].(string)
+	} else if nodeProps["apigroup"] != nil && nodeProps["apigroup"].(string) != "" {
+		groupVersion = "/" + nodeProps["apigroup"].(string)
+	} else if nodeProps["apiversion"] != nil && nodeProps["apiversion"].(string) != "" {
+		groupVersion = "/" + nodeProps["apiversion"].(string)
+		root = "api"
+	}
+
+	selfLink := fmt.Sprintf("/%s%s%s/%s/%s", root, groupVersion, namespace, resourceString, nodeProps["name"].(string))
```

Comment:
Please add an example here to show how the string will be formatted.

Reply:
done